app/src/main/java/woowacourse/paint/ColorUiModel.kt

```diff
@@ -0,0 +1,25 @@
+package woowacourse.paint
+
+import android.content.Context
+import android.graphics.Color
+import androidx.core.content.ContextCompat
+
+data class ColorUiModel(
+    val color: Int,
+) {
+    companion object {
+        val DEFAULT_COLOR = ColorUiModel(Color.RED)
+
+        fun palletColors(context: Context): List<ColorUiModel> =
+            context.resources.obtainTypedArray(R.array.pallet_colors).run {
+                val colors = (0 until length()).map { compatColorOf(context, getResourceId(it, 0)) }
+                recycle()
+                colors
+            }.map { ColorUiModel(it) }
```

Comment:
저는 scope 함수를 사용하는 것보다 변수명을 잘 지어서 사용하는 게 좀 더 가독성이 좋다고 생각하는 편입니다!

이건 지극히 제 개인 취향이라 변경을 요구하는 것은 아니고 이렇게 바꿔볼 수도 있겠다 싶어서 코멘트 남깁니다!
그래서 위 코드를 수정해보면 아래와 같이 수정해볼 수 있을 것 같아용

```suggestion
         fun palletColors(context: Context): List<ColorUiModel> {
            val colorResIds = context.resources.obtainTypedArray(R.array.pallet_colors)
            val colors =
                List(colorResIds.length()) {
                    ColorUiModel(compatColorOf(context, colorResIds.getResourceId(it, 0)))
                }
            colorResIds.recycle()
            return colors
        }
```

Reply:
넵 감사합니다!! 👍